test_files/promisectorlike/promisectorlike.js

```diff
@@ -5,7 +5,7 @@
 goog.module('test_files.promisectorlike.promisectorlike');
 var module = module || { id: 'test_files/promisectorlike/promisectorlike.ts' };
 /**
- * @param {PromiseConstructorLike} ctorLike
+ * @param {function(new: (!PromiseLike<?>), function(function((undefined|?|!PromiseLike<?>)=): void, function(?=): void): void): ?} ctorLike
```

Comment:
Can't this alias expansion potentially cause us to reference types that aren't in scope?

Reply:
No, because we have `ensureSymbolDeclared` that will emit a `forwardDeclare` if necessary. We currently have both code paths, the ensure declared that introduces forward declares, and then sometimes when TS creates a symbol with an aliasSymbol on it, we use that - but we cannot predict or force that, so it's pretty random.